### PR TITLE
ccls: needs gcc >= 7.2 for C++17

### PR DESCRIPTION
**Pull request by xu-cheng**
https://github.com/Homebrew/linuxbrew-core/issues/12518

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/linuxbrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/linuxbrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?
- [ ] Have you included the output of `brew gist-logs <formula>` of the build failure if your PR fixes a build failure. Please quote the exact error message.

-----

### CODE REVIEW

**Comment by xu-cheng** [comment]
Blocked by https://github.com/Homebrew/brew/issues/6070

File: Formula/ccls.rb [hunk 1]
```diff
@@ -15,6 +15,14 @@ class Ccls < Formula
   depends_on "llvm"
   depends_on :macos => :high_sierra # C++ 17 is required
 
+  # C++17 is required
+  fails_with :gcc => "4"
+  fails_with :gcc => "5"
+  fails_with :gcc => "6"
+  fails_with :gcc => "7" do
+    version "7.1"
```
Comment (sjackman) [inline Formula/ccls.rb]: I haven't seen this construct before. What does it mean? Does it fail with `gcc < 7.1` or `gcc ≤ 7.1`, or something else?

Comment (sjackman) [reply Formula/ccls.rb]: On Linux I believe you'll also need,
```ruby
  depends_on "gcc@8" unless OS.mac?
```

Comment (xu-cheng) [reply Formula/ccls.rb]: The syntax is from the document https://github.com/Homebrew/brew/blob/e18c5c1291e3b2a900d2437a9e993d45034c766b/Library/Homebrew/formula.rb#L2489-L2499

Comment (sjackman) [reply Formula/ccls.rb]: Thanks for the tip! So it sounds like it means fails with `gcc ≤ 7.1`. That's helpful.

Comment (sjackman) [reply Formula/ccls.rb]: I miss `needs :cxx17`. 😢

File: Formula/ccls.rb [hunk 2]
```diff
@@ -15,7 +15,22 @@ class Ccls < Formula
...
+  end
+
   def install
+    # https://github.com/Homebrew/brew/issues/6070
+    unless OS.mac?
+      ENV.remove %w[LDFLAGS LIBRARY_PATH HOMEBREW_LIBRARY_PATHS], "#{HOMEBREW_PREFIX}/lib"
+    end
```
Comment (xu-cheng) [inline Formula/ccls.rb]: @sjackman For simplicity, I choose to handle https://github.com/Homebrew/brew/issues/6070 inside the formula for the time being.

Comment (sjackman) [reply Formula/ccls.rb]: That's okay for the time being. Are you interested in working on a PR for a fix to https://github.com/Homebrew/brew/issues/6070 ?

Comment (xu-cheng) [reply Formula/ccls.rb]: I don’t think I have enough free time for the time being.

Comment (sjackman) [reply Formula/ccls.rb]: Fair enough

**Comment by xu-cheng** [comment]
Could we merge this for now?

**Comment by iMichka** [comment]
Thanks @xu-cheng !

**Pull request closed without merging**